core/src/androidMain/kotlin/BluetoothDevice.kt

```diff
@@ -0,0 +1,72 @@
+package com.juul.kable
+
+import android.annotation.TargetApi
+import android.bluetooth.BluetoothDevice
+import android.bluetooth.BluetoothDevice.PHY_LE_1M_MASK
+import android.bluetooth.BluetoothDevice.TRANSPORT_AUTO
+import android.content.Context
+import android.os.Build
+import android.os.Handler
+import android.os.HandlerThread
+import com.juul.kable.gatt.Callback
+import kotlinx.coroutines.android.asCoroutineDispatcher
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.newSingleThreadContext
+
+internal fun BluetoothDevice.connect(
+    context: Context,
+    state: MutableStateFlow<State>,
+): Connection? =
+    if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
+        connectApi26(context, state)
+    } else {
+        connectApi21(context, state)
+    }
+
+@TargetApi(Build.VERSION_CODES.LOLLIPOP)
+private fun BluetoothDevice.connectApi21(
+    context: Context,
+    state: MutableStateFlow<State>,
+): Connection? {
+    val callback = Callback(state)
+    val bluetoothGatt = connectGatt(context, false, callback) ?: return null
+
+    // Kludge: Explicitly set Connecting state so when Peripheral is suspending until Connected, it doesn't incorrectly
+    // see Disconnected before the connection request has kicked off the Connecting state (via Callback).
+    state.value = State.Connecting
```

Comment:
I like seeing the comment on this, just honestly not sure if it's kludge -- seems super reasonable to me

Reply:
I felt it was kludge because it's so disjointed from every other `State` propagation.
Right now, `Callback` which acts as the receiver of messages from Android, just converts the callback style state to the `Flow` `State`...**except** for this one particular instance.

Though, I don't expect to ever change this, so I think you're right that "kludge" probably is a mislabelling and the comment is sufficient to explain _why_ it's being done; it can be changed later and the comment will help guide considerations when/if changing it.

👍